### PR TITLE
feat(perception_dataset/kognic/): Edge cases fix Kognic converter

### PR DESCRIPTION
**Pull request by natanaels12**
## Description

Improve the reliability of the T4 ↔ Kognic conversion pipeline when sensor files, timestamps, point-cloud layouts, or annotations are ambiguous or inconsistent.

- dabc33b — Indexes source lidar timestamps from LIDAR_CONCAT_INFO so per-sensor OpenLABEL frames can be matched to their fused T4 sample by capture timestamp.
- 861f273 — Selects the lidar sample_data record whose timestamp exactly matches the T4 sample timestamp.
- 34f3f4a — Validates annotation_hz as an integer from 1 through 10.
- 1563c58 — Prevents multiple downloaded annotations for one scene from overwriting each other.
- 2f256dd — Reconciles category indices in the T4 dataset’s category.json when bounding-box and semantic-segmentation annotations are imported together.
- 45bb4c6:
    - Removes existing camera and lidar output directories before conversion to prevent stale files from entering the new staging data.
    - Requires every staged camera and lidar sensor to contain the same number of files.
    - Derives the point stride from LIDAR_CONCAT_INFO when available.
    - Validates that per-sensor ranges cover the concatenated point cloud without gaps or overlaps.
    - Supports sensors that contribute zero points by producing a header-only CSV.
    - Rejects point-cloud reads whose byte count does not match the declared source range.
- b738bc5 — Removes the unused drop_camera_token_not_found option. Missing camera frames always receive blank black images.
- 23c120b — Checks camera timestamps against lidar timestamps before pairing sensor files and allows the intended synchronization tolerance.
- 49cd036 — Handles a frame containing only one stream-tagged LiDAR segmentation in kognic annotation as per-sensor data, even though T4 has `LIDAR_CONCAT_INFO`. 
- 906971e — Waits for each Kognic pre-annotation to reach indexed before creating an input. Failed, unknown, missing, and timed-out states raise an error.
- 0ca08fd — Rejects malformed RLE values and point counts larger than the target cloud before expansion.
- 6a1425c — Raise error when multiple images are referring to the same timestamp.
- b835d7f — Requires `lidar_point_stride` to be explicitly configured for fused point clouds that do not have `LIDAR_CONCAT_INFO`. This removes heuristic stride detection, which could interpret the same binary data using multiple plausible point layouts and produce an incorrect point count.
- 7bca248 — Removed `target_hz`, which is a fallback and mostly unused but may cause bugs. 

## How to review
1. Review `perception_dataset/utils/pointcloud.py `for point-cloud stride detection, concat-range validation, empty sensor handling, and short-read checks.
2. Review `t4_to_kognic_converter.py `and upload_dataset.py for stale-file cleanup and sensor frame validation.
3. Check that pre-annotations reach indexed before they are attached to an input.
4. Review openlabel_to_t4_converter.py for annotation ordering, LiDAR timestamp matching, concat-slice reconstruction, and RLE validation.
5. Review the configuration changes for lidar_point_stride and the removal of drop_camera_token_not_found.
6. Run the edge-case tests and confirm they cover the changed behavior.

### test data

Created temporary synthetic data to cover the edge cases. 

### test command

```bash
uv run pytest -q \
  tests/kognic/test_upload_edge_cases.py \
  tests/kognic/test_download_edge_cases.py \
  tests/kognic/test_openlabel_import_edge_cases.py \
  tests/kognic/test_t4_staging_edge_cases.py \
  tests/kognic/test_kognic_dataset_conversion.py

### CODE REVIEW

File: docs/tier_iv_t4_extractor_to_kognic.md [hunk 1]
```diff
@@ -298,8 +299,8 @@ conversion:
   input_base: ./data/non_annotated_t4_format
   output_base: ./data/kognic_format
   workers_number: 12
+  lidar_point_stride: 5
```
Comment (jacoblambert) [inline docs/tier_iv_t4_extractor_to_kognic.md:302]: this should not have to be set here and instead obtained from t4devkit

File: tests/kognic/test_download_edge_cases.py
Comment (jacoblambert) [inline tests/kognic/test_download_edge_cases.py:1]: not sure we should have that as part of normal test suite,

**Review by jacoblambert — DISMISSED** [review]
Some minor points I will ask to be addressed in main PR, otherwise OK.

File: docs/tier_iv_t4_extractor_to_kognic.md (hunk 1 above)
Comment (natanaels12) [reply docs/tier_iv_t4_extractor_to_kognic.md:302]: Yes, so as I suspected I didn't use t4-devkit for the lidar bin loader which is why I needed this. Will make a separate PR to introduce t4-devkit and remove this parameter.

File: tests/kognic/test_download_edge_cases.py
Comment (natanaels12) [reply tests/kognic/test_download_edge_cases.py:1]: Agreed, this is too specific to Kognic’s multi-type annotation workflow for the normal test suite. It would be better covered by a separate integration test, so I’ll remove it for now.
